jujugui/static/gui/src/app/views/ghost-deployer-extension.js

```diff
@@ -127,6 +134,36 @@ YUI.add('ghost-deployer-extension', function(Y) {
     },
 
     /**
+      If we're using Juju 1 then we need to deploy a charm Id which has
+      the series defined in the URL. This is not required for Juju 2 as it
+      supports multi-series charms.
+
+      @method _addSeriesToCharmId
+      @param {String} charmId The charm id.
+      @param {String} series The series of the service.
+    */
+    _addSeriesToCharmId: function(charmId, series) {
+      let charmIdParts = charmId.replace('cs:', '').split('/');
```

Comment:
oh yes, definitely! We also need to start putting the 'utils' into logical modules - it's very unwieldy